R/calcGDPFuture.R

```diff
@@ -1,165 +1,122 @@
-#' @describeIn calcGDPPast Get future GDP projections
-#'
-#' @param GDPFuture A string designating the source for the future GDP data. Available sources are:
-#'   \itemize{
-#'     \item "SSPs": IIASA [SSP database](https://tntcat.iiasa.ac.at/SspDb/dsd?Action=htmlpage&page=welcome)
-#'     \item "SSP2EU": Combined SSP2 and Eurostat (for the EU countries) source
-#'     \item "SDPs":
-#'     \item "MI": Missing island dataset
-#'   }
-#'   See the "Combining data sources with '-'" section below for how to combine data sources.
-calcGDPFuture <- function(GDPFuture, unit) { # nolint
+#' @rdname calcGDPPast
+#' @order 3
+#' @inheritParams calcScenarioConstructor
+calcGDPFuture <- function(futureData = "SSPs") {
   # Check user input
   toolCheckUserInput("GDPFuture", as.list(environment()))
   # Call calcInternalGDPFuture function the appropriate number of times (map) and combine (reduce)
   # !! Keep formula syntax for madrat caching to work
-  purrr::pmap(list("GDPFuture" = unlist(strsplit(GDPFuture, "-")), "unit" = unit),
+  purrr::pmap(list("futureData" = unlist(strsplit(futureData, "-"))),
               ~calcOutput("InternalGDPFuture", aggregate = FALSE, supplementary = TRUE, ...)) %>%
-    toolReduce(mbindOrFillWith = "fillWith")
+    toolListFillWith()
 }
 
-
-######################################################################################
-# Internal Function
-######################################################################################
-calcInternalGDPFuture <- function(GDPFuture, unit) { # nolint
+calcInternalGDPFuture <- function(futureData) {
   data <- switch(
-    GDPFuture,
-    "SSPs"   = calcOutput("InternalGDPFutureSSPs", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    "SSP2"   = calcOutput("InternalGDPFutureSSP2", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    "SSP2EU" = calcOutput("InternalGDPFutureSSP2EU", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    "SDPs"   = calcOutput("InternalGDPFutureSDPs", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    "MI"     = calcOutput("InternalGDPMI", unit = unit, aggregate = FALSE, supplementary = TRUE),
-    stop("Bad input for calcGDPFuture. Invalid 'GDPFuture' argument.")
+    futureData,
+    "SSPs"   = readSource("SSP", "gdp"),
+    "SSP2"   = readSource("SSP", "gdp", "SSP2"),
+    "SSP2EU" = setNames(readSource("SSP", "gdp", "SSP2"), "SSP2EU"),
```

Comment:
this still exists?

Reply:
Yes, but only as a placeholder. Just so that the input data pipeline isn't broken.